kafka-rest/src/main/java/io/confluent/kafkarest/controllers/NoSchemaRecordSerializer.java

```diff
@@ -58,11 +58,15 @@
   }
 
   private static ByteString serializeBinary(JsonNode data) {
+    if (!data.isTextual()) {
+      throw new BadRequestException(String.format("data=%s is not a base64 string.", data));
+    }
```

Comment:
If using `JsonNode`, jackson will never map to `BinaryNode`, since it can't distinguish between a base64 encoded byte-array and a normal string. We use `JsonNode` because the request object is the same for all kinds of requests. We use `BinaryNode` on the places that accept only binary, e.g. `ProduceRequestHeader`.